File: lib/op/interpreter/opcall/params/param/file/validator.test.js
const objectUnderTest = require('./validator')

describe('validate', () => {
  describe('value falsy', () => {
    it('returns expected result', () => {
      /* arrange */
      const expectedResult = [new Error('file required')]

      /* act */
      const actualResult = objectUnderTest.validate('')

      /* assert */
      expect(actualResult).toEqual(expectedResult)
    })
  })
  describe('value truthy', () => {
    it('returns expected result', () => {
      /* arrange/act */
      const actualResult = objectUnderTest.validate('some file')

      /* assert */
      expect(actualResult).toEqual([])
    })
  })
})

File: sdks/go/opspec/interpreter/call/op/params/param/validator_test.go
package param

import (
	"errors"
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
	"github.com/opctl/opctl/sdks/go/model"
)

var _ = Context("Validate", func() {
	objectUnderTest := NewValidator()
	Context("invoked w/ nil param", func() {
		It("should return expected error", func() {
			/* arrange */
			expectedErrs := []error{errors.New("required")}

			/* act */
			actualErrs := objectUnderTest.Validate(
				nil,
				&model.Param{},
			)
			Expect(actualErrs).To(Equal(expectedErrs))
		})
	})
})
